infrastructure/terraform/main.tf

```diff
@@ -0,0 +1,68 @@
+resource "azurerm_resource_group" "main" {
+  name     = local.resource_group_name
+  location = local.region
+}
+
+module "app-key-vault" {
+  source = "../modules/dtos-devops-templates/infrastructure/modules/key-vault"
+
+  name                                             = "kv-${var.app_short_name}-${var.environment}"
+  resource_group_name                              = azurerm_resource_group.main.name
+  enable_rbac_authorization                        = true # TODO: make true by default?
+  location                                         = local.region
+  log_analytics_workspace_id                       = module.log_analytics_workspace_audit.id
+  monitor_diagnostic_setting_keyvault_enabled_logs = ["AuditEvent", "AzurePolicyEvaluationDetails"] # TODO: not required if not creating NSG
+  monitor_diagnostic_setting_keyvault_metrics      = ["AllMetrics"]
+  private_endpoint_properties = { # TODO: Some could be default?
+    private_dns_zone_ids_keyvault        = [data.azurerm_private_dns_zone.key-vault.id]
+    private_endpoint_enabled             = true
+    private_endpoint_subnet_id           = module.container_app_subnet.id
+    private_endpoint_resource_group_name = azurerm_resource_group.main.name
+    private_service_connection_is_manual = false
+  }
+  soft_delete_retention = 7 # TODO: Allow disabling soft delete
+}
+
+module "log_analytics_workspace_audit" {
+  source = "../modules/dtos-devops-templates/infrastructure/modules/log-analytics-workspace"
+
+  name     = "law-${var.environment}-${var.app_short_name}"
+  location = local.region
+
+  law_sku        = "PerGB2018"
```

Comment:
Wonder if this is something that could be defaulted in the module as well. I guess it comes down to which bits you want us to be responsible for, and which should be more of a platform concern?

Reply:
This makes sense, added to my todo list. Once the module is updated, we can update manage